File: iniciante/src/1015.cpp
/* title: Distância Entre Dois Pontos
 * number of the problem: 1015
 * date: 31/08/2021
 * author: ltsdw
 */

#include <iostream>
#include <iomanip>
#include <cmath>

double distance(const double& x1, const double& y1, const double& x2, const double& y2)
{
    return sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
}

int main(int argc, const char* argv[])
{
    using std::cin, std::cout, std::fixed, std::setprecision;

    double x1, y1, x2, y2;
    cin >> x1 >> y1 >> x2 >> y2;

    cout << fixed << setprecision(4) << distance(x1, y1, x2, y2) << '\n';

    return EXIT_SUCCESS;
}

File: haskell/iniciante/1015.hs
-- title: Distância Entre Dois Pontos
-- number of the problem: 1015
-- date: 21/09/20
-- author: ltsdw

import Text.Printf

main :: IO ()
main = do
    l1 <- getLine
    l2 <- getLine

    let (x1 : y1 : _) = map (read :: String -> Double) $ words l1
    let (x2 : y2 : _) = map (read :: String -> Double) $ words l2

    printf "%.4f\n" (sqrt (((x2 - x1) * (x2 - x1)) + ((y2 - y1) * (y2 - y1))))

